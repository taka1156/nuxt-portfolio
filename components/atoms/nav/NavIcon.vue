<template functional>
  <div class="nav-icon">
    <span
      class="nav-icon__top"
      :class="[props.isOpen ? 'nav-icon__top--open' : 'nav-icon__top--close']"
    ></span>
    <span
      class="nav-icon__middle"
      :class="{ 'nav-icon__middle--fade': props.isOpen }"
    ></span>
    <span
      class="nav-icon__bottom"
      :class="[props.isOpen ? 'nav-icon__bottom--open' : 'nav-icon__bottom--close']"
    ></span>
  </div>
</template>

<style scoped>
.nav-icon {
  position: relative;
  top: 0;
  right: 0;
  height: 50px;
  width: 50px;
  cursor: pointer;
}

span {
  display: block;
  height: 2px;
  width: 70%;
  margin: auto;
  position: absolute;
  top: calc((100% - 4px) / 2);
  left: 0;
  right: 0;
  background-color: #fff;
  border-radius: 1px;
}

.nav-icon__top--close {
  transform: translateY(-10px);
}

.nav-icon__bottom--close {
  transform: translateY(10px);
}

.nav-icon__top--open {
  animation: top 0.6s ease;
  transform: rotate(-45deg);
}

.nav-icon__middle--fade {
  animation: middle 0.6s ease;
  opacity: 0;
}

.nav-icon__bottom--open {
  animation: bottom 0.6s ease;
  transform: rotate(45deg);
}

@keyframes top {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(45deg);
  }
}

@keyframes middle {
  from {
    opacity: 1;
  }

  to {
    opacity: 0;
  }
}

@keyframes bottom {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(-45deg);
  }
}
</style>
