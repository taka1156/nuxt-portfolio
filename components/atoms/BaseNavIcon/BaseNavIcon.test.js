import { shallowMount } from '@vue/test-utils';
import NavIcon from './BaseNavIcon.vue';

// NavIcon
describe('NavIcon', () => {
  const navIcon = propsData => {
    return shallowMount(NavIcon, {
      propsData: {
        ...propsData,
      },
    });
  };

  it('開いた時(閉じるアイコン表示)', () => {
    const wrapper = navIcon({ isOpen: true });
    const TOP = wrapper.find('.nav-icon__top');
    const MIDDLE = wrapper.find('.nav-icon__middle');
    const BOTTOM = wrapper.find('.nav-icon__bottom');
    expect(TOP.classes()).toContain('nav-icon__top--open');
    expect(MIDDLE.classes()).toContain('nav-icon__middle--fade');
    expect(BOTTOM.classes()).toContain('nav-icon__bottom--open');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('閉じる時(開くアイコン表示)', () => {
    const wrapper = navIcon({ isOpen: false });
    const TOP = wrapper.find('.nav-icon__top');
    const MIDDLE = wrapper.find('.nav-icon__middle');
    const BOTTOM = wrapper.find('.nav-icon__bottom');
    expect(TOP.classes()).toContain('nav-icon__top--close');
    expect(MIDDLE.classes()).not.toContain('nav-icon__middle--fade');
    expect(BOTTOM.classes()).toContain('nav-icon__bottom--close');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
