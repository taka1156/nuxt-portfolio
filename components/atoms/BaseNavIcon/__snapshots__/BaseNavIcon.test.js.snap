// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseNavIcon 閉じる時(開くアイコン表示) 1`] = `"<div class=\\"base-nav-icon base-nav-icon--extend\\"><button><span class=\\"base-nav-icon__border base-nav-icon__top base-nav-icon__top--close\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__middle\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__bottom base-nav-icon__bottom--close\\"></span> <span class=\\"base-nav-icon__text\\">NAVI</span></button></div>"`;

exports[`BaseNavIcon 開いた時(閉じるアイコン表示) 1`] = `"<div class=\\"base-nav-icon base-nav-icon--extend\\"><button><span class=\\"base-nav-icon__border base-nav-icon__top base-nav-icon__top--open\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__middle base-nav-icon__middle--fade\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__bottom base-nav-icon__bottom--open\\"></span> <span class=\\"base-nav-icon__text\\">NAVI</span></button></div>"`;
